### PR TITLE
Pass arguments to host function as const Value*

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

**Comment by chfast** [comment]
```
fizzy/execute/blake2b/512_bytes_rounds_1_mean                     -0.0466         -0.0466            87            83            87            83
fizzy/execute/blake2b/512_bytes_rounds_16_mean                    -0.0469         -0.0469          1316          1254          1316          1254
fizzy/execute/ecpairing/onepoint_mean                             -0.0572         -0.0572        410735        387251        410738        387254
fizzy/execute/keccak256/512_bytes_rounds_1_mean                   -0.0288         -0.0288           101            98           101            98
fizzy/execute/keccak256/512_bytes_rounds_16_mean                  -0.0370         -0.0370          1503          1447          1503          1447
fizzy/execute/memset/256_bytes_mean                               -0.1201         -0.1201             7             6             7             6
fizzy/execute/memset/60000_bytes_mean                             -0.1262         -0.1262          1599          1397          1599          1397
fizzy/execute/mul256_opt0/input1_mean                             -0.1268         -0.1268            29            25            29            25
fizzy/execute/ramanujan_pi/33_runs_mean                           -0.0470         -0.0470           131           125           131           125
fizzy/execute/sha1/512_bytes_rounds_1_mean                        -0.1235         -0.1235            94            83            94            83
fizzy/execute/sha1/512_bytes_rounds_16_mean                       -0.1257         -0.1257          1314          1149          1314          1149
fizzy/execute/sha256/512_bytes_rounds_1_mean                      -0.1704         -0.1704           100            83           100            83
fizzy/execute/sha256/512_bytes_rounds_16_mean                     -0.1655         -0.1655          1367          1141          1367          1141
fizzy/execute/taylor_pi/pi_1000000_runs_mean                      -0.0126         -0.0126         40540         40028         40541         40028
fizzy/execute/micro/eli_interpreter/exec105_mean                  -0.0957         -0.0957             5             4             5             4
fizzy/execute/micro/factorial/20_mean                             -0.0808         -0.0808             1             1             1             1
fizzy/execute/micro/fibonacci/24_mean                             -0.0652         -0.0652          5230          4889          5230          4889
fizzy/execute/micro/host_adler32/1_mean                           -0.0139         -0.0139             0             0             0             0
fizzy/execute/micro/host_adler32/1000_mean                        -0.0088         -0.0088            29            29            29            29
fizzy/execute/micro/spinner/1_mean                                -0.0751         -0.0751             0             0             0             0
fizzy/execute/micro/spinner/1000_mean                             -0.1015         -0.1015            10             9            10             9
```

**Comment by axic** [comment]
The `ecpairing` change which should not be affected just shows we're still experiencing the same measurement/optimisation problem.

File: lib/fizzy/execute.cpp [hunk 1]
```diff
@@ -486,7 +486,7 @@ inline bool invoke_function(
 {
     const auto num_args = func_type.inputs.size();
     assert(stack.size() >= num_args);
-    span<const Value> call_args{stack.rend() - num_args, num_args};
+    const auto call_args = stack.rend() - num_args;
```
Comment (axic) [inline lib/fizzy/execute.cpp:489]: Didn't I do the same change in some PR already? Perhaps one of your old refactorings.

**Review by axic — APPROVED** [review]
I'd wait for @gumb0's review too.

File: lib/fizzy/capi.cpp [hunk 2]
```diff
@@ -68,9 +68,9 @@ inline fizzy::ExecutionResult unwrap(const FizzyExecutionResult& result) noexcep
 
 inline auto unwrap(FizzyExternalFn func, void* context) noexcept
 {
-    return [func, context](fizzy::Instance& instance, fizzy::span<const fizzy::Value> args,
+    return [func, context](fizzy::Instance& instance, const fizzy::Value* args,
                int depth) noexcept -> fizzy::ExecutionResult {
-        const auto result = func(context, wrap(&instance), wrap(args.data()), args.size(), depth);
+        const auto result = func(context, wrap(&instance), wrap(args), 0, depth);
```
Comment (chfast) [inline lib/fizzy/capi.cpp:73]: This should be fixed, now or later.

Comment (gumb0) [reply lib/fizzy/capi.cpp]: Pushed the fix.

**Review by gumb0 — APPROVED** [review]

**Review by axic — APPROVED** [review]

**Pull request merged**